src/Control/Monad/Morph.hs

```diff
@@ -116,6 +120,54 @@ class MFunctor t where
     -}
     hoist :: (Monad m) => (forall a . m a -> n a) -> t m b -> t n b
 
+{-| Convert a Monad to an isomorphic transformation of Identity
+> mtrans (a >> b)   = mtrans a >> mtrans b
+-}
+class Monad m => MTrans t m where
+    mtrans :: m a -> t Identity a
```

Comment:
Would it be worth including the opposite direction of the isomorphism as a separate method in thi same class?

Reply:
Thanks for your comments. I'm not sure whether to include the opposite direction, I suppose so but I'm insufficiently expert to know if it's a good idea; I'm still uncertain about my current suggestion.

Reply:
@code5hot: If it truly is an isomorphism then you probably want to include the opposite direction, if only to prove that it is an isomorphism.  The hardest part is mainly deciding what to name both directions.  Maybe `toTrans` and `fromTrans`?

Reply:
I suppose this operation is about ensuring there is transformish structure and relaxing that requirement - but we only want to guarantee there's an inverse, we don't want to guarantee the result isn't a transformer, so it's not really conversion and so I don't think toTrans and fromTrans are appropriate.

maybe openTrans and closeTrans ?

Reply:
oh, there might be other tranformer concepts apart from monads, so:

openMTStack and closeMTStack ?

Reply:
Okay, these are the only correct names I can think of. They're correct because even if 'm' is already 't Identity' they still tell the truth, it's a stack after toStack and it's a monad after toMonad. They're also co-operative because they don't assume that monad transformer stacks are the only types of transformer stacks or that transformer stacks are the only type of monad stacks (which is untrue in both cases, I think).

monadToTStack :: m a -> t Identity a
stackToTMonad :: t Identity a -> m a